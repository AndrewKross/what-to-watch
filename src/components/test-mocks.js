export const PromoFilmData = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`
};

const GENRE = `drama`;
const RELEASE = `2020`;
const RATING = 9;
const VOTES = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://cdn.videvo.net/videvo_files/video/free/2019-09/small_watermarked/190828_07_MarinaBayatNightDrone_UHD_02_preview.webm`;
const RUN_TIME = 140;

const DESCRRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`
];

const review = {
  id: `0`,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date()
};

export const films = TITLES.map((title, index) => {
  return ({
    id: index.toString(),
    title,
    cover: COVERS[index],
    poster: COVERS[index],
    preview: PREVIEW,
    genre: GENRE,
    release: RELEASE,
    rating: RATING,
    votes: VOTES,
    description: DESCRRIPTION,
    director: DIRECTOR,
    actors: ACTORS,
    runTime: RUN_TIME,
    reviews: [review]
  });
});

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`
];
