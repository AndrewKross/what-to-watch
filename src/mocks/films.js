import { nanoid } from "nanoid";
import {
  getRandomIntegerNumber,
  getRandomArrayItem,
  getRandomArrayItems,
} from "../utils/common.js";
import { reviews } from "./reviews.js";

const ID_LENGTH = 8;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`,
];

const PREVIEWS = [
  `https://cdn.videvo.net/videvo_files/video/free/2019-09/small_watermarked/190828_07_MarinaBayatNightDrone_UHD_02_preview.webm`, `https://www.videvo.net/videvo_files/converted/2018_07/preview/180607_A_064.mp429860.webm`, `https://www.videvo.net/videvo_files/converted/2018_01/preview/171124_H1_006.mp491073.webm`, `https://www.videvo.net/videvo_files/converted/2016_11/preview/GOPR6239_1.mov34724.webm`, `https://cdn.videvo.net/videvo_files/video/free/2019-12/small_watermarked/190915_B_01_Timelapse%20Danang_01_preview.webm`,
];

const GENRES = [
  `comedy`,
  `drama`,
  `melodrama`,
  `horror`,
  `crime`,
  `documentary`,
  `romance`,
  `thriller`,
];

const DESCRRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const DIRECTORS = [
  `Otto Bathurst`,
  `Tom Harper`,
  `Colm McCarthy`,
  `Tim Mielants`,
  `David Caffrey`,
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`,
  `Sophie Rundle`,
  `Joe Cole`,
  `Finn Cole`,
];

const ReleaseRange = {
  MIN: 1980,
  MAX: 2020,
};

const RatingRange = {
  MIN: 0,
  MAX: 10,
};

const ratingsCountRange = {
  MIN: 0,
  MAX: 1000,
};

const RunTimeRange = {
  MIN: 20,
  MAX: 240,
};

export const filmsData = TITLES.map((title, index) => {
  return {
    id: nanoid(ID_LENGTH),
    title,
    cover: COVERS[index],
    poster: COVERS[index],
    preview: getRandomArrayItem(PREVIEWS),
    genre: getRandomArrayItem(GENRES),
    release: getRandomIntegerNumber(
      ReleaseRange.MIN,
      ReleaseRange.MAX
    ).toString(),
    rating: +(Math.random() * RatingRange.MAX).toPrecision(2),
    ratingsCount: getRandomIntegerNumber(
      ratingsCountRange.MIN,
      ratingsCountRange.MAX
    ),
    description: DESCRRIPTION,
    director: getRandomArrayItem(DIRECTORS),
    actors: getRandomArrayItems(ACTORS),
    runTime: getRandomIntegerNumber(RunTimeRange.MIN, RunTimeRange.MAX),
    reviews,
  };
});
