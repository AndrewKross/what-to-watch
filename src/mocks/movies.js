const imgFolder = `./img/`;

export default [
  {
    id: `film-1`,
    title: `Revenant`,
    poster: imgFolder + `revenant.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Action`,
  },
  {
    id: `film-2`,
    title: `War of the Worlds`,
    poster: imgFolder + `war-of-the-worlds.jpg`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Sci-Fi`,
  },
  {
    id: `film-3`,
    title: `Snatch`,
    poster: imgFolder + `snatch.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedy`,
  },
  {
    id: `film-4`,
    title: `Shutter Island`,
    poster: imgFolder + `shutter-island.jpg`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Thriller`,
  },
  {
    id: `film-5`,
    title: `Pulp Fiction`,
    poster: imgFolder + `pulp-fiction.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`,
  },
];
