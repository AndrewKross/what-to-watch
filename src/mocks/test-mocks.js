import { NameSpace } from '../reducer/reducer';
import { DEFAULT_GENRE, LoadingStatus } from '../const';

const RELEASE = 2020;
const RATING = 9;
const RATINGS_COUNT = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`;
const RUN_TIME = 140;
const COMMENTS_COUNT = 6;

const DESCRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`,
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`,
];

export const GENRES = [
  `all genres`,
  `comedy`,
  `drama`,
  `melodrama`,
  `horror`,
  `crime`,
  `documentary`,
  `romance`,
  `thriller`,
];

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`,
];

const comment = {
  id: 0,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date(`December 25, 2019 01:00:00`),
};

export const comments = Array(COMMENTS_COUNT)
  .fill(``)
  .map(() => comment);

export const films = TITLES.map((title, index) => {
  return ({
    id: index,
    title,
    cover: COVERS[index],
    posterImage: COVERS[index],
    previewImage: COVERS[index],
    backgroundImage: COVERS[index],
    previewVideo: PREVIEW,
    videoMain: PREVIEW,
    genre: GENRES[index + 1] || GENRES[1],
    released: RELEASE,
    rating: RATING,
    ratingsCount: RATINGS_COUNT,
    description: DESCRIPTION,
    director: DIRECTOR,
    starring: ACTORS,
    runTime: RUN_TIME,
    isFavorite: false,
    backgroundColor: `black`,
  });
});

export const userData = {
  id: 1,
  email: `Andrew@gmail.com`,
  name: `Andrew`,
  avatar: `img/1.png`,
};

export const loginData = {
  email: `Andrew@gmail.com`,
  password: `1234`,
};

export const reviewToPost = {
  rating: 8,
  comment: `Great!`,
};

export const defaultMockStore = {
  [NameSpace.DATA]: {
    films,
    comments,
    isFilmsLoaded: true,
    isCommentsLoaded: true,
  },
  [NameSpace.APP_STATE]: {
    filmsOnScreen: 8,
    currentGenre: DEFAULT_GENRE,
  },
  [NameSpace.USER]: {
    isAuthorized: true,
    userInfo: userData,
    reviewLoadingStatus: LoadingStatus.PENDING,
    authorizationLoadingStatus: LoadingStatus.PENDING,
  },
};
