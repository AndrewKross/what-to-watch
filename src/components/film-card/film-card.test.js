import React from "react";
import renderer from "react-test-renderer";
import { films } from "../../mocks/test-mocks";
import FilmCard from "./film-card.jsx";

const [film] = films;

test(`FilmCard should render correctly`, () => {
  const tree = renderer
    .create(
      <FilmCard
        filmData={film}
        onFilmCardHover={() => { }}
        onFilmCardClick={() => { }}
      />, {
        createNodeMock: () => ({}),
      },
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
