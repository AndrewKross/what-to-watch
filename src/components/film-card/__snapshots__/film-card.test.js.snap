// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard should render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height={175}
      muted={true}
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      src="https://cdn.videvo.net/videvo_files/video/free/2019-09/small_watermarked/190828_07_MarinaBayatNightDrone_UHD_02_preview.webm"
      width={280}
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="film-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
